phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -2203,6 +2196,89 @@ public MetaDataResponse call(MetaDataService instance) throws IOException {
                 return rpcCallback.get();
             }
         });
+
+        //For view creation, if SYSCAT rpc succeeds, mark child_link rows as VERIFIED
+        if (!childLinkMutations.isEmpty()) {
+            try {
+                if (result.getMutationCode() == MutationCode.TABLE_NOT_FOUND) {
+                    sendChildLinkMutations(childLinkMutations, true, false, physicalTableNameBytes, schemaBytes);
+                } else {
+                    sendChildLinkMutations(childLinkMutations, true, true, physicalTableNameBytes, schemaBytes);
+                }
+            }
+            catch (SQLException e) {
+                //unverified rows will be repaired during read
+                LOGGER.debug("Exception in phase-3 of view creation: " + e.getMessage());
+                addChildLinkScanTask();
+            }
+        }
+        return result;
+    }
+
+    /*
+    Helper method to send mutations to SYSTEM.CHILD_LINK using its endpoint coprocessor
+     */
+    public void sendChildLinkMutations(List<Mutation> mutations, boolean isVerified, boolean isDelete,
+                                        byte[] physicalTableNameBytes, byte[] schemaBytes)
+            throws SQLException {
+
+        // get empty column information
+        PTable childLinkLogicalTable = getTable(null, PhoenixDatabaseMetaData.SYSTEM_CHILD_LINK_NAME, HConstants.LATEST_TIMESTAMP);
+        byte[] emptyCF = SchemaUtil.getEmptyColumnFamily(childLinkLogicalTable);
+        byte[] emptyCQ = EncodedColumnsUtil.getEmptyKeyValueInfo(childLinkLogicalTable).getFirst();
+
+        // add empty column value to mutations or create delete mutations for phase-3
+        List<Mutation> childLinkMutations = new ArrayList<>();
+        for (Mutation m : mutations) {
+            if (isDelete) {
+                Delete delete = new Delete(m.getRow());
+                childLinkMutations.add(delete);
+            }
+            else {
+                Put put = isVerified ? new Put(m.getRow()) : (Put)m;
+                byte[] emptyColumnValue = isVerified ? VERIFIED_BYTES : UNVERIFIED_BYTES;
+                put.addColumn(emptyCF, emptyCQ, IndexUtil.getMaxTimestamp(m), emptyColumnValue);
```

Comment:
When you are updating the empty column to verified shouldn't that be a different timestamp. There should be two versions of the empty column cell. One with unverified and other with verified and verified one should have a later timestamp. Can you verify this ?

Reply:
During debugging, I had used dumpTable and did see the verified empty column mutation with a higher timestamp.

Reply:
But you are setting the timestamp from the existing cells in the mutation `IndexUtil.getMaxTimestamp(m)`. I am surprised you are getting a higher timestamp.

Reply:
Dump from one of the tests:
```
\x00\x00T_N000001\x00\x00V_N000003/0:LINK_TYPE/1687542526662/Put/vlen=1/seqid=0 column= LINK_TYPE val=\x04
\x00\x00T_N000001\x00\x00V_N000003/0:_0/1687542526683/Put/vlen=1/seqid=0 column= _0 val=\x01
\x00\x00T_N000001\x00\x00V_N000003/0:_0/1687542526662/Put/vlen=1/seqid=0 column= _0 val=\x02
```

Reply:
If I am reading this correctly, your verified version of the empty column will have the same timestamp as the unverified version.

Reply:
Yes, you are right. I printed out all timestamps seen in `IndexUtil.getMaxTimestamp(m)` and for the test, they are always long max value. So these timestamps are being assigned at the server side?